docs/data/data-grid/editing/ValidateRowModelControlGrid.tsx

```diff
@@ -15,6 +16,39 @@ function validateEmail(email) {
 }
 
 export default function ValidateRowModelControlGrid() {
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    visibleFields: VISIBLE_FIELDS,
+    rowLength: 5,
+  });
+  const columns: GridColumns = [
+    { field: 'name', headerName: 'Name', width: 180, editable: true },
+    {
+      field: 'email',
+      headerName: 'Email',
+      width: 200,
+      editable: true,
+      preProcessEditCellProps: (params: GridPreProcessEditCellProps) => {
+        const isValid = validateEmail(params.props.value);
+        return { ...params.props, error: !isValid };
+      },
+    },
+    {
+      field: 'dateCreated',
+      headerName: 'Date Created',
+      type: 'date',
+      width: 180,
+      editable: true,
+    },
+    {
+      field: 'lastLogin',
```

Comment:
The employee dataset doesn't this column. You can use `lastUpdated` to fake it.

```suggestion
      field: 'lastUpdated',
```